File: backend/src/main/java/com/flowlink/core/graph/GraphEdge.java
package com.flowlink.core.graph;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * 노드 간 연결. fromPort는 분기 노드(if)의 'true'/'false' 또는 기본 'out'.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record GraphEdge(
        String id,
        String from,
        String fromPort,
        String to
) {
    public String fromPortOrDefault() {
        return fromPort == null || fromPort.isBlank() ? "out" : fromPort;
    }
}

File: backend/src/main/kotlin/com/flowlink/core/graph/GraphEdge.kt
package com.flowlink.core.graph

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

/**
 * 노드 간 연결. fromPort는 분기 노드(if)의 'true'/'false' 또는 기본 'out'.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
data class GraphEdge(
    val id: String?,
    val from: String?,
    val fromPort: String?,
    val to: String?
) {
    fun fromPortOrDefault(): String =
        if (fromPort == null || fromPort.isBlank()) "out" else fromPort
}
